### PR TITLE
rbac: migrate to newer api-version 2018-01-01-preview to support data actions

### PR DESCRIPTION
**Pull request by yugangw-msft**
There are a few breaking changes caused by the SDK, but they are relatively trivial and actually make the command better and easier to use, hence i just go ahead...

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5805
This is an **experimental preview** for @microsoft users.

**Comment by yugangw-msft** [comment]
@lmazuel , let me know when new azure-mgmt-authorization gets published, so i will adjust this PR by removing the private wheel

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py [hunk 1]
```diff
@@ -223,11 +221,10 @@ def build_msi_role_assignment(cmd, vm_vmss_name, vm_vmss_resource_id, role_defin
 
     # pylint: disable=line-too-long
     msi_rp_api_version = '2015-08-31-PREVIEW'
-    authorization_api_version = get_mgmt_service_client(cmd.cli_ctx, AuthorizationManagementClient).api_version
     return {
         'name': name,
         'type': assignment_type,
-        'apiVersion': authorization_api_version,
+        'apiVersion': '2015-07-01',  # the minimum api-version to create the assignment
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: this change is mainly for azure stack as the 2018-01-01-preview is bit too new

File: src/azure-cli-testsdk/azure/cli/testsdk/base.py [hunk 2]
```diff
@@ -131,6 +131,18 @@ def create_random_name(self, prefix, length):
 
         return moniker
 
+    def create_guid(self):
```
Comment (yugangw-msft) [inline src/azure-cli-testsdk/azure/cli/testsdk/base.py]: This should fix the long standing issue that playback will not work if the command create a guid which can be part of url

File: src/azure-cli-testsdk/azure/cli/testsdk/base.py [hunk 3]
```diff
@@ -131,6 +131,18 @@ def create_random_name(self, prefix, length):
...
+        import uuid
+        self.test_guid_count += 1
+        moniker = '88888888-0000-0000-0000-00000000' + ("%0.4X" % self.test_guid_count)
```
Comment (troydai) [inline src/azure-cli-testsdk/azure/cli/testsdk/base.py]: Is this `8` just a lucky number?

File: src/azure-cli-testsdk/azure/cli/testsdk/base.py (hunk 2 above)
Comment (troydai) [reply src/azure-cli-testsdk/azure/cli/testsdk/base.py]: Can you add comments to the method to explain the use case for this method? The name of the method is too generic to indicate its usage.

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/latest/test_role.py [hunk 4]
```diff
@@ -40,10 +45,7 @@ def test_create_for_rbac_with_secret_with_assignment(self, resource_group):
         })
 
         try:
-            guids = ['88DAAF5A-EA86-4A68-9D45-477538D41300', '88DAAF5A-EA86-4A68-9D45-477538D41301',
-                     '88DAAF5A-EA86-4A68-9D45-477538D41302', '88DAAF5A-EA86-4A68-9D45-477538D41303',
-                     '88DAAF5A-EA86-4A68-9D45-477538D41304', '88DAAF5A-EA86-4A68-9D45-477538D41305']
-            with self.get_guid_gen_patch(guids):
+            with mock.patch('azure.cli.command_modules.role.custom._gen_guid', side_effect=self.create_guid):
```
Comment (troydai) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/latest/test_role.py]: +1

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/latest/test_role.py [hunk 5]
```diff
@@ -285,7 +286,7 @@ def test_role_assignment_audits(self, resource_group):
                     self.cmd('role assignment list-changelogs --start-time {}'.format(start_time))
                 else:
                     # NOTE: get the time range from the recording file and use them below for playback
-                    start_time, end_time = '2018-02-10T21:36:06Z', '2018-02-11T21:39:05Z'
+                    start_time, end_time = '2018-03-11T18:43:08Z', '2018-03-12T18:46:29Z'
```
Comment (troydai) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/latest/test_role.py]: Ugr, no other way? This seems to make re-record difficult.

Comment (yugangw-msft) [reply src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/latest/test_role.py]: I will open an issue and address it separately once i know more about the filter pattern

**Pull request merged**